python/huggingfaceserver/tests/test_vlllm_model.py

```diff
@@ -0,0 +1,2898 @@
+    #                             " ears": -1.3619931936264038,
+    #                             " paws": -2.2743258476257324,
+    #                         },
+    #                         {
+    #                             " with": -4.619960308074951,
+    #                             " and": -0.805719792842865,
+    #                             ",": -1.6155686378479004,
+    #                         },
+    #                         {
+    #                             " fluffy": -5.6248779296875,
+    #                             " a": -1.4977400302886963,
+    #                             " tiny": -3.006150484085083,
+    #                         },
+    #                         {
+    #                             " white": -2.2152767181396484,
+    #                             " fur": -1.9012728929519653,
+    #                         },
+    #                     ],
+    #                 ),
+    #                 text="- Labrador! He has tiny ears with fluffy white",
+    #             )
+    #         ],
+    #         created=1719569921,
+    #         model="opt-125m",
+    #         system_fingerprint=None,
+    #         object="text_completion",
+    #         usage=CompletionUsage(
+    #             completion_tokens=10, prompt_tokens=7, total_tokens=17
+    #         ),
+    #     )
+    #
+    #     assert response.id == expected.id
+    #     assert response.choices == expected.choices
+    #     assert response.system_fingerprint == expected.system_fingerprint
+    #     assert response.object == expected.object
+    #     assert response.usage == expected.usage
+    #     assert response.model == expected.model
+    #     assert isinstance(response.created, int)
+
+    async def test_vllm_completion_facebook_opt_model_with_logprobs_stream(
+        self, vllm_opt_model
+    ):
+        opt_model, mock_vllm_engine = vllm_opt_model
+        request_id = "cmpl-d771287a234c44498e345f0a429d6691"
+        model_name = "opt-125m"
+
+        async def mock_generate(*args, **kwargs) -> AsyncIterator[RequestOutput]:
+            for cmpl_chunk in opt_cmpl_chunks_with_logprobs:
+                cmpl_chunk.request_id = args[2]
+                yield cmpl_chunk
+
+        mock_vllm_engine.generate = mock_generate
+
+        prompt = "Hi, I love my cat"
+        params = CreateCompletionRequest(
+            model=model_name,
+            prompt=prompt,
+            stream=True,
+            max_tokens=10,
+            logprobs=2,
+        )
+        request = CompletionRequest(request_id=request_id, params=params, context={})
+        response_iterator = await opt_model.create_completion(request)
+        completion = ""
+        log_probs = Logprobs(
+            text_offset=[], token_logprobs=[], tokens=[], top_logprobs=[]
+        )
+        async for resp in response_iterator:
+            assert resp.id == request_id
+            completion += resp.choices[0].text
+            assert len(resp.choices) == 1
```

Comment:
Should we assert on length before doing `completion += resp.choices[0].text`?

Reply:
done